core/src/main/java/com/linecorp/armeria/server/annotation/ExceptionHandling.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface ExceptionHandling {
```

Comment:
How about just adding a method that matches an exception to `ExceptionHandlingFunction`?

```java
public interface ExceptionHandlingFunction {
    default boolean accept(Throwable cause) {
        return true;
    }
    ...
}
```

Than we wouldn't this need annotation.

Reply:
Do you mean that we use ExceptionHandlingFunction list instead of map when finding a suitable exception handler?

Reply:
Yes. Even if we specified `@ExceptionHandling(SomeException.class)` we will have to do some iteration of class hierarchy and interface traversal anyway. Given that the number of exception handler is going to be small, I think it may be fine. What do you think?

Reply:
Actually this was another option that I thought. So I think it's also okay. :-)